tests/raftstore/cluster.rs

```diff
@@ -397,7 +418,15 @@ impl<T: Simulator> Cluster<T> {
 
     pub fn shutdown(&mut self) {
         debug!("about to shutdown cluster");
-        let keys = self.sim.rl().get_node_ids();
+        let keys;
+        match self.sim.try_read() {
+            Ok(s) => keys = s.get_node_ids(),
+            Err(sync::TryLockError::Poisoned(e)) => {
+                let s = e.into_inner();
```

Comment:
I met panic on panic and test core dump. beacause test case panic and run drop will call shutdown, and the Lock reason will panic again. this check to prevent it and catch the first panic stack information.